### DISCUSSION TITLE
WPSD and HamClock

### DISCUSSION THREAD

**Discussion by Steve-WB4IZC** [VA3HDL/hamdashboard#23] [Ideas]
Has any thought been given to being able to display one's HamClock live stream feed and/ or a live caller WPSD dashboard page in one of the 12 windows?  
![WPSD DStar](https://github.com/user-attachments/assets/62b42c05-49f3-4d77-b278-a853d1172dd7)
![HamClock live](https://github.com/user-attachments/assets/08c6431a-10b9-4301-8de5-12bf96b6a1d6)

**Comment by VA3HDL** [comment]
Hi Steve,,

I started working on this but I got nowhere. I'm planning to go back and get it done. It has been requested few times already.

73,
Pablo

**Reply by Steve-WB4IZC** [reply]
Thank you for responding.  I've been trying a few things also but, so far I can't seem to get it working.  Much appreciated when you do find time to look at this again.  I run three WPSD hotspots and it would be nice to display each "live caller" formatted dashboard in a row of windows.  In any case thanks again for such a great program.  de WB4IZC

**Reply by midair55** [reply]
I sorta have a solution…..

Reference FAQ #42 at https://www.clearskyinstitute.com/ham/HamClock

I wrote a one line script to create a bmp screen shot of the Hamclock using the curl command and Restful parameter as shown in the FAQ’s examples.  The script is executed every 30 seconds using two crontab jobs.   The Hamclock panel on the dashboard points to the bmp file and refreshes every 30 seconds.

So…. Not realtime Hamclock but close….

**Reply by VA3HDL** [reply]
> I sorta have a solution…..
> 
> Reference FAQ #42 at https://www.clearskyinstitute.com/ham/HamClock
> 
> I wrote a one line script to create a bmp screen shot of the Hamclock using the curl command and Restful parameter as shown in the FAQ’s examples. The script is executed every 30 seconds using two crontab jobs. The Hamclock panel on the dashboard points to the bmp file and refreshes every 30 seconds.
> 
> So…. Not realtime Hamclock but close….

This can be done on the dashboard without any scripts or cronjobs using this URL:
 "http://hamclock_IP:8080/get_capture.bmp"

**Reply by Steve-WB4IZC** [reply]
I did this and yes I was able to display the captured bmp file but, I was not able to get it to update even though I inserted the line four times and had it rotating through the four lines every seven seconds.

**Reply by VA3HDL** [reply]
> I did this and yes I was able to display the captured bmp file but, I was not able to get it to update even though I inserted the line four times and had it rotating through the four lines every seven seconds.

The problem is the browser cache. For some images I can use "anti-cache" mechanism in the code, but it didn't work with the Hamclock. It just throws an error. Check setting a couple different maps in Hamclock so the map changes time to time.

**Reply by midair55** [reply]
I changed mine to:  http://192.168.1.133:9000/get_capture.bmp and it doesn't update due to the cache.   Pointing to the captured bmp file doesn't have that problem using Chrome.  I haven't tried other browsers.

**Reply by midair55** [reply]
To beat this horse a bit more.....  My HAM dashboard is:  https://qsl.net/n5ng/HAM
I also have it on a RPi on my LAN.  The Hamclock bmp file is generated on the RPi and pushed to the qsl.net server every minute via FTP.  I'm avoiding a shorter interval since the bmp file is 3M small and I don't want the qsl.net guys getting upset with the amount of file transfer traffic.   The Hamclock panel refreshes every minute.

I plan to do a port forward on my router so the bmp image file can be fetched from my RPi to avoid the FTPs to the qsl.net server but that's for a rainy day....

Side note:  My Weather centric dashboard is:  https://qsl.net/n5ng

**Reply by VA3HDL** [reply]
> To beat this horse a bit more..... My HAM dashboard is: https://qsl.net/n5ng/HAM I also have it on a RPi on my LAN. The Hamclock bmp file is generated on the RPi and pushed to the qsl.net server every minute via FTP. I'm avoiding a shorter interval since the bmp file is 3M small and I don't want the qsl.net guys getting upset with the amount of file transfer traffic. The Hamclock panel refreshes every minute.
> 
> I plan to do a port forward on my router so the bmp image file can be fetched from my RPi to avoid the FTPs to the qsl.net server but that's for a rainy day....
> 
> Side note: My Weather centric dashboard is: https://qsl.net/n5ng

Your dashboards are amazing! Thanks for sharing!

**Comment by Steve-WB4IZC** [comment]
Thank you for the update.  Yes, I think that will work for my need.  I appreciate your suggestion.  73, Steve WB4IZC

**Comment by VA3HDL** [comment]
Hi @Steve-WB4IZC 

The last two releases include the ability to show live websites on any of the tiles as an iFrame.
You must use the keyword "iFrame|" before the URL link for that tile.

Thanks.
